### PR TITLE
Fix test_cppcheck.py

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Fix arguments of cppcheck test and some cppcheck warnings (version 1.87)

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
On my machine with cppcheck version 1.87 I'm getting many cppcheck warnings. They are mainly about shadows functions and variables and style warnings.
I started fixing the warnings in the src file and only it remains only these about using std built-in functions for loops:

```
[/home/stefi/develop/webots/src/wren/PostProcessingEffect.hpp:213]: (style) Consider using std::find_if algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Transform.hpp:63]: (style) Consider using std::transform algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Transform.hpp:75]: (style) Consider using std::transform algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/DynamicMesh.cpp:545]: (style) Consider using std::transform algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Frustum.cpp:71]: (style) Consider using std::any_of algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Frustum.cpp:80]: (style) Consider using std::any_of algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/FrameBuffer.cpp:79]: (style) Consider using std::any_of, std::all_of, std::none_of, or std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Material.cpp:47]: (style) Consider using std::any_of, std::all_of, std::none_of, or std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Material.cpp:98]: (style) Consider using std::any_of, std::all_of, std::none_of, or std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Material.cpp:151]: (style) Consider using std::count_if algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/PbrMaterial.cpp:80]: (style) Consider using std::fill or std::generate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/PbrMaterial.cpp:83]: (style) Consider using std::fill or std::generate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/PhongMaterial.cpp:62]: (style) Consider using std::fill or std::generate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/PhongMaterial.cpp:65]: (style) Consider using std::fill or std::generate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Primitive.cpp:346]: (style) Consider using std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/ShadowVolumeCaster.cpp:173] -> [/home/stefi/develop/webots/src/wren/ShadowVolumeCaster.cpp:182]: (style) Local variable normal shadows outer variable
[/home/stefi/develop/webots/src/wren/ShadowVolumeCaster.cpp:264] -> [/home/stefi/develop/webots/src/wren/ShadowVolumeCaster.cpp:274]: (style) Local variable normal shadows outer variable
[/home/stefi/develop/webots/src/wren/Skeleton.cpp:74]: (style) Consider using std::find_if algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Skeleton.cpp:85]: (style) Consider using std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Texture.cpp:173]: (style) Consider using std::any_of algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Texture2d.cpp:57]: (style) Consider using std::count_if algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/TextureTransform.cpp:92]: (style) Consider using std::any_of algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/Transform.cpp:53]: (style) Consider using std::accumulate algorithm instead of a raw loop.
[/home/stefi/develop/webots/src/wren/StaticMesh.cpp:1528]: (style) Consider using std::count_if algorithm instead of a raw loop.
```

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
:+1:

File: projects/vehicles/controllers/autonomous_vehicle/autonomous_vehicle.c [hunk 1]
```diff
@@ -281,9 +281,9 @@ void update_display() {
 
 void compute_gps_speed() {
   const double *coords = wb_gps_get_values(gps);
-  const double speed = wb_gps_get_speed(gps);
+  const double spees_ms = wb_gps_get_speed(gps);
```
Comment (omichel) [inline projects/vehicles/controllers/autonomous_vehicle/autonomous_vehicle.c]:
```suggestion
  const double speed_ms = wb_gps_get_speed(gps);
```

File: projects/vehicles/controllers/autonomous_vehicle/autonomous_vehicle.c [hunk 2]
```diff
@@ -281,9 +281,9 @@ void update_display() {
...
   // store into global variables
-  gps_speed = speed * 3.6;  // convert from m/s to km/h
+  gps_speed = spees_ms * 3.6;  // convert from m/s to km/h
```
Comment (omichel) [inline projects/vehicles/controllers/autonomous_vehicle/autonomous_vehicle.c]:
```suggestion
  gps_speed = speed_ms * 3.6;  // convert from m/s to km/h
```

**Comment by fabienrohrer** [comment]
> What do others think about it?

I'm quite neutral with this topic. I have a small preference to use the `std` loops. I don't see any issue with `std` loops and they are now fundamentally implemented in C++. I'm also not very familiar with this, but it's not a big deal to me to adopt them, and I'm glad to learn new C++ features.

**Comment by DavidMansolino** [comment]
I also think they are not very simple to read, for loops it is ok, but something like this:
```
std::accumulate(mChildren.begin(), mChildren.end(), mChildren.size(), [](int sum, const Node *n) { return sum + n->computeChildCount(); });
```

Is in my point of view less readable than:
```
int count = mChildren.size();
for (Node *n : mChildren)
  count += n->computeChildCount();
```

But that's probably because we are not used to these functions.

**Comment by stefaniapedrazzi** [comment]
This is what cppcheck suggests to do, but we can suppress all the stl algorithm warnings.

The first time it is necessary to read the docs to really understand  how these functions works, and then for simplicity I also used the lambda function that we never used before.
Note that these functions are very similar to the Javascript Array functions "forEach", "some", etc.

**Comment by omichel** [comment]
I personally agree with the comment of @DavidMansolino. The only point that would convince me to use std algorithms would be the CPU performance, but it doesn't seem obvious that the std algorithms are more performant than the loops. Do we have any hint on that?

**Comment by stefaniapedrazzi** [comment]
Here is the link to an article explaining the introduction of these warnings:
http://pfultz2.com/blog/2018/11/04/stl-alogrithms/

**Comment by stefaniapedrazzi** [comment]
I don't have any preferences about the two solutions.
The built-in algorithms do exactly what we want to achieve with the raw loop even if they seem to be less readable.

For me it is just important to be consistent and use the STL algorithm everywhere when possible or suppress the warnings.

**Comment by stefaniapedrazzi** [comment]
I don't have any preferences about the two solutions.
The built-in algorithms do exactly what we want to achieve with the raw loop even if they seem to be less readable.

For me it is just important to be consistent and use the STL algorithm everywhere when possible or suppress the warnings.

**Comment by omichel** [comment]
OK, so I would tend to give priority to readability versus supposed performance. I believe that performance should be addressed if a bottleneck is identified. So, if nobody objects, I would prefer that we drop this STL algorithms warning and revert back to the more readable loops. Also, performance-wise, I believe that we have more control with loops where we could later decide to use openmp to parallelize them if needed, whereas STL algorithms don't offer this level of control and remain a black box.

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
test me

**Comment by stefaniapedrazzi** [comment]
:+1:

**Pull request merged**